### PR TITLE
Cloud API: Add Fedora IoT bootable container

### PR DESCRIPTION
**Pull request by achilleas-k**
Add the Fedora iot-bootable-container to the cloud API and include it in the API tests.
See https://github.com/osbuild/images/pull/361 for image definition introduction.

We don't intend to have this available in osbuild-composer on-prem (Weldr), so it's added to the deny list.

-----

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by ondrejbudai — DISMISSED** [review]
:heart:

**Review by thozza — APPROVED** [review]
LGTM

**Review by thozza — DISMISSED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -536,6 +536,12 @@ API:
         RUNNER:
           - aws/fedora-38-x86_64
           - aws/fedora-39-x86_64
+      - IMAGE_TYPE: ["iot-bootable-container"]
+        RUNNER:
+          - aws/fedora-39-x86_64
```
Comment (ondrejbudai) [inline .gitlab-ci.yml:541]: Ftr, this is targetting F40+. I know that supporting F39 is nice to have, but if this proves to be impossible, I'm happy with dropping it.

**Review by ondrejbudai — DISMISSED** [review]
:heart:

File: .gitlab-ci.yml (hunk 1 above)
Comment (achilleas-k) [reply .gitlab-ci.yml:541]: The image builds fine on F39 in my own testing (and in osbuild/images) and bootc is available there so it should be fine.  If any issues come up that are only solvable on F40 I guess we can drop it.

**Comment by achilleas-k** [comment]
Cherry-picked the two commits from https://github.com/osbuild/osbuild-composer/pull/3919#pullrequestreview-1851861995 for CI fixage.

**Pull request merged**